data/tutorials/lg_08_error_handling.md

```diff
@@ -125,67 +191,549 @@ let () =
     )
 ```
 
-Each printer should take care of the exceptions it knows about, returning
-`Some <printed exception>`, and return `None` otherwise (let the other printers
-do the job!).
+Each printer should take care of the exceptions it knows about, returning `Some
+<printed exception>`, and return `None` otherwise (let the other printers do the
+job!).
+
+## Runtime Crashes
+
+Although OCaml is a very safe language, it is possible to trigger unrecoverable
+errors at runtime.
+
+### Exceptions Not Raised
+
+Under panic circumstances, the native code compiler gives its best effort for
+raising meaningful exceptions. However, some error conditions may remain
+undetected, which will result in a segmentation fault. This is the specially the
+case for stack overflows, which aren't always detected.
+
+> But catching stack overflows is tricky, both in Unix-like systems and under
+> Windows, so the current implementation in OCaml is a best effort that is
+> occasionally buggy.
+
+[Xavier Leroy, October
+2021](https://discuss.ocaml.org/t/stack-overflow-reported-as-segfault/8646/8)
+
+### Bypassing Type-Safety
+
+OCaml provides means to bypass its type safety. Don't use it. Here is how
+to shoot oneself in the foot:
+
+```shell
+> echo "(Obj.magic () : int array).(0)" > foo.ml
+> ocamlopt foo.ml
+> ./a.out
+Segmentation fault (core dumped)
+```
+
+### Language Bugs
+
+When a crash isn't coming from:
+* A limitation of the native code compiler
+* `Obj.magic`
+
+It may be a language bug. It happens. Here is what to do when this is suspected:
+
+1. Make sure the crash affects both compilers: bytecode and native
+1. Write a self-contained and minimal proof-of-concept code which does nothing
+   but triggering the crash
+1. File an issue in the [OCaml Bug Tracker in
+   GitHub](https://github.com/ocaml/ocaml/issues)
+
+Here is an example of such a bug: https://github.com/ocaml/ocaml/issues/7241
+
+### Safe vs. Unsafe Functions
+
+Uncaught exceptions raise runtime crashes. Therefore, there is a tendency to use
+the following terminology:
+* Function raising exceptions: Unsafe
+* Function handling errors in data: Safe
+
+The main means to write such kind of safe error handling functions is to use
+either `Option` (next section) or `Result` (following section).
+
+## Using the `Option` Type for Errors
+
+The `Option` module provides the first alternative to exceptions. The `'a
+option` datatype allows to express either the availability of data for instance
+`Some 42` or the absence of data using `None`, which can represent an error.
+
+Using `Option` it is possible to write functions that return `None` instead of
+throwing an exception.
+```ocaml
+let div_opt m n =
+  try Some (m / n) with
+    Division_by_zero -> None
+
+let find_opt p l =
+  try Some (List.find p l) with
+    Not_found -> None
```

Comment:
Should there be an `;;` at the end here?
If I put it there, I get this response: 
```
val div_opt : int -> int -> int option = <fun>
val find_opt : ('a -> bool) -> 'a list -> 'a option = <fun>
```